### PR TITLE
Add docs, specs and some pattern matchings

### PR DESCRIPTION
**Pull request by aleDsz**
Based on issue #4 , I just organized the modules and generate the specs and documentation of both the functions and the modules themselves, already preparing to generate docs for HexDocs, including adding ExDoc as a dependency so you can generate and have a preview of the documentation.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+  @version "0.1.0"
+  @repo "https://github.com/samuelpordeus/typo_killer"
+
   def project do
     [
       app: :typo_killer,
-      version: "0.1.0",
-      elixir: "~> 1.9",
+      version: @version,
+      elixir: "~> 1.8",
```
Comment (samuelpordeus) [inline mix.exs]: hey! why exactly we are changing elixir version to `1.8`? `dialyxir`? 🙂

File: lib/typo_killer.ex [hunk 2]
```diff
@@ -1,9 +1,23 @@
 defmodule TypoKiller do
+  @moduledoc """
+  Qu'est-ce que c'est
+
+  Fa-fa-fa-fa-fa-fa-fa-fa-fa
+
+  It's a funny app to search wrong typos inside projects and report
+  the words that can be fixed and, with this project, you can
+  contribute to open-source repos with the famous `Fix typo` commits/PRs.
```
Comment (samuelpordeus) [inline lib/typo_killer.ex]: nice description here ✨

File: lib/typo_killer.ex [hunk 3]
```diff
@@ -1,9 +1,23 @@
 defmodule TypoKiller do
+  @moduledoc """
+  Qu'est-ce que c'est
+
+  Fa-fa-fa-fa-fa-fa-fa-fa-fa
+
+  It's a funny app to search wrong typos inside projects and report
```
Comment (samuelpordeus) [inline lib/typo_killer.ex]:
```suggestion
  It's a funny app to search typos inside projects and report
```

Every typo is wrong 💀 😆

File: mix.exs (hunk 1 above)
Comment (aleDsz) [reply mix.exs]: Oops, i just forgot to update, because my environment is for Elixir 1.8 >_<
I'll change

File: lib/typo_killer.ex (hunk 3 above)
Comment (aleDsz) [reply lib/typo_killer.ex]: HAHAHAHA, okay
wait a sec

File: lib/typo_killer/file_parser.ex [hunk 4]
```diff
@@ -1,4 +1,16 @@
 defmodule TypoKiller.FileParser do
+  @moduledoc """
+  Parse all files and concat paths to folder to generate a complete list of files ready to be read
```
Comment (samuelpordeus) [inline lib/typo_killer/file_parser.ex]:
```suggestion
  Parse all files and concat path to a folder to generate a complete list of files ready to be read
```

path here? 🤔

File: mix.exs (hunk 1 above)
Comment (aleDsz) [reply mix.exs]: Maybe i can add ASDF `.tool-versions` file to use Elixir `1.9`

File: lib/typo_killer/finder.ex [hunk 5]
```diff
@@ -1,7 +1,15 @@
 defmodule TypoKiller.Finder do
+  @moduledoc """
+  Find typos based on percentage and try to avoid the max of possible false positives
+  """
   @minimum_bag_distance 0.75
   @minimum_jaro_distance 0.95
 
+  @doc """
+  Based on a list of words, generate possible typos with an English dictionary
```
Comment (samuelpordeus) [inline lib/typo_killer/finder.ex]:
```suggestion
  Based on a list of words, search for possible typos by comparing them with an auxiliar dictionary
```

**Review by samuelpordeus — APPROVED** [review]
just added some minor comments, thanks for the PR. it's fantastic ✨

File: mix.exs (hunk 1 above)
Comment (samuelpordeus) [reply mix.exs]: @aleDsz sounds like a good plan

**Comment by aleDsz** [comment]
Updated with requested changes

**Comment by msramos** [comment]
Thank you so much @aleDsz ❤️💚💛💙

**Pull request merged**